### ISSUE TITLE
PdCore:javadoc task fails

### ISSUE THREAD

**Issue by joebowbeer** [libpd/pd-for-android#28]
This prevents a clean bintrayUpload from working. Remove javadoc artiffact and related tasks?

````
>gradlew PdCore:javadoc
:PdCore:javadoc
PdCore\src\org\puredata\android\io\AudioWrapper.java:12: error: cannot find symbol
import org.puredata.android.service.R;
                                   ^
  symbol:   class R
  location: package org.puredata.android.service
PdCore\src\org\puredata\android\midi\MidiToPdAdapter.java:12: error: package com.noisepages.netto
yeur.midi does not exist
import com.noisepages.nettoyeur.midi.MidiReceiver;
                                    ^
PdCore\src\org\puredata\android\midi\MidiToPdAdapter.java:19: error: cannot find symbol
public class MidiToPdAdapter implements MidiReceiver {
                                        ^
  symbol: class MidiReceiver
PdCore\src\org\puredata\android\midi\PdToMidiAdapter.java:12: error: package com.noisepages.netto
yeur.midi does not exist
import com.noisepages.nettoyeur.midi.MidiReceiver;
                                    ^
PdCore\src\org\puredata\android\midi\PdToMidiAdapter.java:21: error: cannot find symbol
        private final MidiReceiver receiver;
                      ^
  symbol:   class MidiReceiver
  location: class PdToMidiAdapter
PdCore\src\org\puredata\android\midi\PdToMidiAdapter.java:29: error: cannot find symbol
        public PdToMidiAdapter(MidiReceiver receiver) {
                               ^
  symbol:   class MidiReceiver
  location: class PdToMidiAdapter
PdCore\src\org\puredata\android\service\PdService.java:33: error: package android.support.v4.app
does not exist
import android.support.v4.app.NotificationCompat;
````

**Comment by tkirshboim** [comment]
These are warnings and I don't think it's the reason why the BintrayUpload task (partly) fails.
The task creates the javadoc and that can be read. The warnings are from external classes are not resolved properly and in those cases the javadoc refers to the Object class, which is not complete but better than having no javadoc at all.. Also,  since JCenter publishes only OSS it might be mandatory to include documentation next to the sources, but I'm not sure about that.
We could also try for the next releases use the Android Studio 'Tools > Generate JavaDoc' option to see if it does it better.

**Comment by joebowbeer** [comment]
The javadoc task did/does fail, when I ran it separately or as part of `bintrayUpload`. When run on behalf of bintrayUpload, its failure caused bintrayUpload to fail -- before it failed for some other reason.

There should be a way to restrict the sources or ignore errors.

This blog is the most comprehensive source I can locate:

[Building Javadocs for Android library projects with Gradle](http://mbcdev.com/2015/02/18/building-javadocs-for-android-library-projects-with-gradle/)

**Comment by tkirshboim** [comment]
Can you post the output of the gradle command? When I run ```./gradlew clean BintrayUpload``` this is the result at the end:

```
:PdCore:bintrayUpload FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':PdCore:bintrayUpload'.
> Could not upload to 'https://api.bintray.com/content/pd-for-android/maven/pd-for-android/1.0.0-rc3/org/puredata/android/pd-core/1.0.0-rc3/pd-core-1.0.0-rc3-javadoc.jar': HTTP/1.1 409 Conflict [message:Unable to upload files: An artifact with the path 'org/puredata/android/pd-core/1.0.0-rc3/pd-core-1.0.0-rc3-javadoc.jar' already exists]

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```
At fails because the file already exists, which makes sense..

**Comment by joebowbeer** [comment]
Below is the javadoc failure:

````
./gradlew PdCore:javadoc

[...]

1 error
56 warnings
:PdCore:javadoc FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':PdCore:javadoc'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): '/pd-for-android/PdCore/build/tmp/javadoc/javadoc.options'
````

**Comment by joebowbeer** [comment]
However, of the many 'errors' and warnings listed in the output, there is only one show-stopper:

````
/pd-for-android/PdCore/jni/libpd/java/org/puredata/core/utils/IoUtils.java:35: error:
@param name not found
   * @param name of the resulting file
````

Fixing this problem allows javadoc to run.

**Comment by joebowbeer** [comment]
A pull request to fix the javadoc comment in libpd: https://github.com/libpd/libpd/pull/111

**Comment by joebowbeer** [comment]
Assigning to @tkirshboim to merge the pull request.

**Comment by tkirshboim** [comment]
I merged it, however this is still a mystery to me what exactly is happening.
I checked out the master branch before the merge and still could not get the same result like you did. I had 10 warnings but the task succeeds:
```
./gradlew PdCore:javadoc
:PdCore:javadoc
[...]
10 warnings

BUILD SUCCESSFUL
```
Something must different in our setup at the moment I don't have an idea what that could be which could cause these differences.
